File: programmers/weekly/week2.py
def solution(scores):
    answer = ''
    for x in range(len(scores)):
        total = 0;
        isMin = True;
        isMax = True;
        for y in range(len(scores)):
            if x != y:
                if scores[x][x] <= scores[y][x]: isMax = False;
                if scores[x][x] >= scores[y][x]: isMin = False;
                total += scores[y][x]
        if isMax or isMin:
            avg = total / (len(scores) - 1)
        else:
            avg = (total + scores[x][x]) / len(scores)
        if avg >= 90: answer += 'A';
        elif avg >= 80: answer += 'B';
        elif avg >= 70: answer += 'C';
        elif avg >= 50: answer += 'D';
        else: answer += 'F'
        
    return answer

File: programmers/weekly/week2.js
function solution(scores) {
    var answer = '';
    for(let x = 0; x < scores.length; x ++){
        let isMax = true;
        let isMin = true;
        let avg;
        let total = 0;
        for(let y = 0; y < scores.length; y++){
            if(y !== x){
               if(scores[x][x] <= scores[y][x]) isMax = false;
               if(scores[x][x] >= scores[y][x]) isMin = false;
               total += scores[y][x]
           }
        }
        if(isMax || isMin) avg = total / (scores.length-1);
        else {
            total += scores[x][x];
            avg = total / scores.length
        }
        if(avg >= 90) answer += 'A';
        else if(avg >= 80) answer += 'B';
        else if(avg >= 70) answer += 'C';
        else if(avg >= 50) answer += 'D';
        else answer += "F";
    }
    return answer;
}